waterlog/src/Pages/ComponentLeakage.js

```diff
@@ -0,0 +1,40 @@
+import React, { Component }from 'react';
+import PropTypes from 'prop-types';
+import { connect } from 'react-redux';
+import { fetchSegmentsLeaksById } from '../actions/SegmentLeaksByIdActions';
+import LeakageInfoComponent from './LeakageInfoComponent';
+import WastageSummaryComponent from './../Components/WastageSummary';
+
+class ComponentLeakage extends Component {
+    
+    componentWillMount() {
+		this.props.fetchSegmentsLeaksById(this.props.id);
+	}
+
+    render (){
+        if(this.props.leak === {}){
```

Comment:
Use this to check if it is an object:
`(Object.entries(this.props.leak).length === 0 && this.props.leak.constructor === Object)`